pkg/cloud/services/route53/route53.go

```diff
@@ -307,32 +309,36 @@ func (s *Service) describeClusterHostedZone(ctx context.Context) (string, error)
 }
 
 func (s *Service) getIngressIP(ctx context.Context) (string, error) {
-	serviceName := fmt.Sprintf("%s%s", IngressAppPrefix, s.scope.Name())
-
-	o := client.ObjectKey{
-		Name:      serviceName,
-		Namespace: IngressAppNamespace,
-	}
 
 	k8sClient, err := s.scope.ClusterK8sClient(ctx)
 	if err != nil {
 		return "", microerror.Mask(err)
 	}
 
-	var icService corev1.Service
-	err = k8sClient.Get(ctx, o, &icService)
-	// Ingress service is not installed in this cluster.
-	if apierrors.IsNotFound(err) {
-		return "", nil
-	} else if err != nil {
+	listOpts := []client.ListOption{
+		client.InNamespace(IngressAppNamespace),
+		client.MatchingLabels{AppNameLabelKey: IngressAppLabel},
+	}
+
+	icServices := &corev1.ServiceList{}
```

Comment:
```suggestion
	var icServices corev1.ServiceList
```
This is a matter of taste, but I try not to use pointers in a function. Will require changes on other lines if you accept this.